tf-psa-crypto/tests/suites/test_suite_psa_crypto.data

```diff
@@ -7481,6 +7481,10 @@ PSA generate key: RSA, maximum size exceeded
 depends_on:PSA_WANT_ALG_RSA_PKCS1V15_CRYPT:PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_GENERATE
 generate_key:PSA_KEY_TYPE_RSA_KEY_PAIR:PSA_VENDOR_RSA_MAX_KEY_BITS+8:PSA_KEY_USAGE_EXPORT | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_DECRYPT:PSA_ALG_RSA_PKCS1V15_CRYPT:PSA_ERROR_NOT_SUPPORTED:0
 
+PSA generate key: RSA, key pair size does not fit in static key buffer
+depends_on:PSA_WANT_ALG_RSA_PKCS1V15_CRYPT:PSA_WANT_KEY_TYPE_RSA_KEY_PAIR_GENERATE:MBEDTLS_PSA_STATIC_KEY_SLOTS:MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE<2364
```

Comment:
> I need a bit of help here for the CI failures as they are all for Windows targets, but I don't have a Win PC to test them. If I'm not wrong the failure is
`273>suites/main_test.function(862): fatal error C1017: invalid integer constant expression [c:\windows\workspace\mbed-tls-pr-head_PR-9302-head\worktrees\tmpcd18d9ao\cmake_solution\tf-psa-crypto\tests\test_suite_psa_crypto.vcxproj]`. This might be connected to `MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE<2364` that I added as dependency in `test_suite_psa_crypto.data`, but then I don't understand why Linux test systems are not failing in that case :thinking: 

Both the fact that this is the only change in this file since a passing CI run, and the line number in the C file, confirm that this line in the culprit. In the C file, the line is 780 lines below `#line 82 "suites/main_test.function"`:

```
#if (MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE<2364)
```

In both the default configuration and the full configuration, `MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE` is not defined in public headers, so the definition from `psa_crypto_core.h` (which is included at that point in `test_suite_psa_crypto.c`) takes effect:
```
#define MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE  \
    (PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE > PSA_CIPHER_MAX_KEY_LENGTH) ? \
    PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE : PSA_CIPHER_MAX_KEY_LENGTH
```

On Windows, we test in the `full` configuration minus a few non-PSA-related options. Both in the default configuration and in the full configuration, on Linux, this macro expands to
```
(((9u * (((4096u) / 2u + 1u) / 8u + 5u) + 14u) > ((((8192u) + 7u) / 8u))) ? (9u * (((4096u) / 2u + 1u) / 8u + 5u) + 14u) : ((((8192u) + 7u) / 8u)) > 32u) ? ((9u * (((4096u) / 2u + 1u) / 8u + 5u) + 14u) > ((((8192u) + 7u) / 8u))) ? (9u * (((4096u) / 2u + 1u) / 8u + 5u) + 14u) : ((((8192u) + 7u) / 8u)) : 32u
```

There's nothing that could be problematic, in particular no comma operator (which GCC and Clang tolerate) or cast.

So I'm stumped as well.

Reply:
Making this change (which isn't right, but shows what the problem is) fixes the build

```
diff --git a/tf-psa-crypto/core/psa_crypto_core.h b/tf-psa-crypto/core/psa_crypto_core.h
index 2a358fc040..a6e1fdf77e 100644
--- a/tf-psa-crypto/core/psa_crypto_core.h
+++ b/tf-psa-crypto/core/psa_crypto_core.h
@@ -59,9 +59,7 @@ typedef enum {
  * set it to the maximum between PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE and
  * PSA_CIPHER_MAX_KEY_LENGTH. */
 #if !defined(MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE)
-#define MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE  \
-    (PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE > PSA_CIPHER_MAX_KEY_LENGTH) ? \
-    PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE : PSA_CIPHER_MAX_KEY_LENGTH
+#define MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE  2000
 #endif /* !MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE*/

 /** The data structure representing a key slot, containing key material
```

Reply:
(I've deleted my previous comment)

Yes, it's the definition of `MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE` in `psa_crypto_core.h` that is the problem, but it's not the ternary operator. Replacing all the macros in that line with constants does make it work... investigating a bit further right now

Reply:
It compiles if I make this one change

```
diff --git a/tf-psa-crypto/core/psa_crypto_core.h b/tf-psa-crypto/core/psa_crypto_core.h
index 2a358fc040..a0380e4b4d 100644
--- a/tf-psa-crypto/core/psa_crypto_core.h
+++ b/tf-psa-crypto/core/psa_crypto_core.h
@@ -61,7 +61,7 @@ typedef enum {
 #if !defined(MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE)
 #define MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE  \
     (PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE > PSA_CIPHER_MAX_KEY_LENGTH) ? \
-    PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE : PSA_CIPHER_MAX_KEY_LENGTH
+    (PSA_EXPORT_KEY_PAIR_OR_PUBLIC_MAX_SIZE) : PSA_CIPHER_MAX_KEY_LENGTH
 #endif /* !MBEDTLS_PSA_STATIC_KEY_SLOT_BUFFER_SIZE*/

 /** The data structure representing a key slot, containing key material
```